src/db/database.ts

```diff
@@ -1,328 +1,297 @@
+      const { fieldType, valueBuf } = res;
+
+      const mps = await this.indexFile.seek(key as string, fieldType);
+      const mp = mps[0];
+      const { fieldType: mpFieldType } = await readIndexMeta(
+        await mp.metadata()
+      );
+
+
+      let ord: "ASC" | "DESC" = "ASC";
+      if (query.orderBy && query.orderBy[0]) {
+        ord = query.orderBy[0].direction
+      }
+
+      const bptree = new BPTree(
+        this.indexFile.getResolver(),
+        mp,
+        dfResolver,
+        format,
+        mpFieldType
+      );
+
+      if (operation === ">") {
+        if (ord === "ASC") {
+          const valueRef = new ReferencedValue({ offset: 0n, length: 0 }, valueBuf)
+          const iter = bptree.iter(valueRef);
+
+          while (await iter.next()) {
+            const currentKey = iter.getKey();
+
+            let res = ReferencedValue.compareBytes(valueBuf, currentKey.value);
+            if (res === -1) {
+              const [_, mp] = await bptree.find(currentKey);
+
+              const data = await this.dataFile.get(
+                Number(mp.offset),
+                Number(mp.offset) + mp.length - 1
+              );
+
+              yield handleSelect(data, query.select);
+            }
+          }
+        } else {
+          const lastKey = await bptree.last();
+          const iter = bptree.iter(lastKey);
+
+          while (await iter.prev()) {
+            const currentKey = iter.getKey();
+
+            let res = ReferencedValue.compareBytes(valueBuf, currentKey.value);
+            if (res === -1) {
+              const [_, mp] = await bptree.find(currentKey);
+
+              const data = await this.dataFile.get(
+                Number(mp.offset),
+                Number(mp.offset) + mp.length - 1
+              );
+              yield handleSelect(data, query.select);
+            } else {
+              console.log("less than")
+            }
+          }
+        }
+      } else if (operation === ">=") {
+        if (ord === "ASC") {
+          const valueRef = new ReferencedValue({ offset: 0n, length: 0 }, valueBuf)
+          const iter = bptree.iter(valueRef);
+
+          while (await iter.next()) {
+            const currentKey = iter.getKey();
+
+            let res = ReferencedValue.compareBytes(valueBuf, currentKey.value);
+            if (res === -1 || res === 0) {
+              const [_, mp] = await bptree.find(currentKey);
+
+              const data = await this.dataFile.get(
+                Number(mp.offset),
+                Number(mp.offset) + mp.length - 1
+              );
+
+              yield handleSelect(data, query.select);
+            }
+          }
+        } else {
+          const lastKey = await bptree.last();
+          const iter = bptree.iter(lastKey);
+
+          while (await iter.prev()) {
+            const currentKey = iter.getKey();
+
+            let res = ReferencedValue.compareBytes(valueBuf, currentKey.value);
+            if (res === -1 || res === 0) {
```

Comment:
```suggestion
            if (ReferencedValue.compareBytes(valueBuf, currentKey.value) <= 0) {
```

Technically, `compareBytes` could return any negative number, it's not guaranteed to be -1.

Reply:
Yeah -- I've updated every single comparison assertion to follow this standard where `<= 0` or `< 0`

For the opposite standard, I've kept it ` >= 0` and `=== 1`